core/controllers/acl_decorators.py

```diff
@@ -462,6 +463,220 @@ def test_can_manage_emails(self, **kwargs):
     return test_can_manage_emails
 
 
+def can_access_blog_admin_page(handler):
+    """Decorator to check whether user can access blog admin page.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now checks if the user has
+        permission to access the blog admin page.
+    """
+
+    def test_can_access_blog_admin_page(self, **kwargs):
+        """Checks if the user is logged in and can access blog admin page.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user does not have credentials to
+                access the blog admin page.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        if role_services.ACTION_ACCESS_BLOG_ADMIN_PAGE in self.user.actions:
+            return handler(self, **kwargs)
+
+        raise self.UnauthorizedUserException(
+            'You do not have credentials to access blog admin page.')
+    test_can_access_blog_admin_page.__wrapped__ = True
+
+    return test_can_access_blog_admin_page
+
+
+def can_manage_blog_post_editors(handler):
+    """Decorator to check whether user can add and remove users as blog
+    post editors.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now checks if the user has
+        permission to manage blog post editors.
+    """
+
+    def test_can_manage_blog_post_editors(self, **kwargs):
+        """Checks if the user is logged in and can add and remove users as blog
+        post editors.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user does not have credentials to
+                manage blog post editors..
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        if role_services.ACTION_MANAGE_BLOG_POST_EDITORS in self.user.actions:
+            return handler(self, **kwargs)
+
+        raise self.UnauthorizedUserException(
+            'You do not have credentials to add or remove blog post editors.')
+    test_can_manage_blog_post_editors.__wrapped__ = True
+
+    return test_can_manage_blog_post_editors
+
+
+def can_access_blog_dashboard(handler):
+    """Decorator to check whether user can access blog dashboard.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now checks if the user has
+        permission to access the blog dashboard.
+    """
+
```

Comment:
Do not leave a newline at the begging of the function definition (We are writing  lint check for the same) [Ditto elsewhere]

Reply:
Done